src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +196,58 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
    if (name.empty())
       return R_UnboundValue;
    
    SEXP env = ns.empty() ? R_GlobalEnv : findNamespace(ns);
+   if (env == R_UnboundValue) return R_UnboundValue;
+   
+   // We might want to use `Rf_findFun`, but it calls `Rf_error` on failure.
```

Comment:
Note that the call to Rf_error isn't a problem if it's wrapped within r::exec::executeSafely